lessons/python_basics/IntroPython.ipynb

```diff
@@ -0,0 +1,1303 @@
+     "text": [
+      "odd\n",
+      "even\n",
+      "odd\n",
+      "even\n",
+      "odd\n",
+      "even\n"
+     ]
+    }
+   ],
+   "source": [
+    "# in this case, 'n' is a dummy variable that will be used by the for loop\n",
+    "# you do not need to assign it ahead of time\n",
+    "\n",
+    "for n in nums:\n",
+    "    if n%2 == 0:\n",
+    "        print('even')\n",
+    "    else:\n",
+    "        print('odd')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "a\n",
+      "e\n",
+      "i\n",
+      "o\n",
+      "u\n"
+     ]
+    }
+   ],
+   "source": [
+    "# for loops can iterate over strings as well\n",
+    "vowels = 'aeiou'\n",
+    "for vowel in vowels:\n",
+    "    print(vowel)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# always use descriptive naming for functions, variables, arguments etc.\n",
+    "def sum_of_squares(num1, num2):\n",
+    "    ss = num1**2 + num2**2\n",
+    "    print(ss)\n",
+    "    return(ss)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "20\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(sum_of_squares(4, 2))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "50\n"
+     ]
+    }
+   ],
+   "source": [
+    "# the return statement in a function allows us to store the output of a function call in a variable for later use\n",
+    "ss1 = sum_of_squares(5,5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 44,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "50\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(ss1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Useful Packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# use a package by importing it, you can also give it a shorter alias, in this case 'np'\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "array = np.arange(15)\n",
+    "lst = range(15)"
```

Comment:
These old comments don't disappear any more if you use the multi-review mode, but I guess it is there now that I look at the final thing.